### PR TITLE
feat(telecom): softphone download link for stores

### PR DESCRIPTION
**Pull request by antonymarion**
ref: MANAGER-13659

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat#MANAGER-13659
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
~~- [ ] Breaking change is mentioned in relevant commits~~

## Description

Give the possibility to users to download app either using QR code or direct link, or badges

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { ALT_SOFTPHONE_STORE_LINK, MOBILE_OS } from './softphone.constants';
+
+export default class SoftphoneController {
+  constructor($window) {
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:4]:
```suggestion
export default class SoftphoneController {
   /* @ngInject */
  constructor($window) {
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 2]
```diff
@@ -2,10 +2,67 @@
...
+        ></strong>
+    </h4>
+
+    <p class="pt-1">
+        <b
+            data-translate="telephony_line_softphone_download_links_option_1"
+        ></b>
+    </p>
+    <div class="row mb-2">
+        <div class="col-md-6 d-flex  justify-content-left align-items-center">
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:17]:
```suggestion
        <div class="col-md-6 d-flex justify-content-left align-items-center">
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.constants.js [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+  windows: 'Windows Phone',
+  android: 'Android',
+  ios: 'iOS',
+};
 
+export const ALT_SOFTPHONE_STORE_LINK = {
+  windows: 'telephony_line_softphone_store_link_windows',
+  android: 'telephony_line_softphone_store_link_android',
+  ios: 'telephony_line_softphone_store_link_ios',
+};
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.constants.js:12]: **nitpick:** No need to maintain it in constants file. Directly use the required string in HTML.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    this.userAgent = $window.navigator.userAgent;
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:5]: **nitpick:** This is not used anywhere. Can you remove it?
```suggestion
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.constants.js [hunk 5]
```diff
@@ -1,5 +1,11 @@
 export const SOFTPHONE_TYPE = 'softphone';
+export const MOBILE_OS = {
+  windows: 'windows',
+  android: 'android',
+  ios: 'ios',
```
Comment (dectotam) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.constants.js:5]: To be consistent with other contants, shouldn't those be in upper case ?

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 6]
```diff
@@ -2,10 +2,70 @@
...
+    <div class="row mb-2">
+        <div class="col-md-6 d-flex justify-content-left align-items-center">
+            <a href="{{$ctrl.storeLinks[$ctrl.MOBILE_OS.ios]}}">
+                <img
+                    alt="{{:: 'telephony_line_softphone_store_link_ios' | translate}}"
+                    src="https://developer.apple.com/app-store/marketing/guidelines/images/badge-example-preferred.png"
+                    width="170"
+                />
+            </a>
+            <a href="{{$ctrl.storeLinks[$ctrl.MOBILE_OS.android]}}">
```
Comment (dectotam) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:25]:
```suggestion
            <a data-ng-href="{{$ctrl.storeLinks[$ctrl.MOBILE_OS.android]}}">
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 7]
```diff
@@ -2,10 +2,70 @@
...
+                <img
+                    alt="{{:: 'telephony_line_softphone_store_link_android' | translate}}"
+                    src="https://play.google.com/intl/en_us/badges/static/images/badges/en_badge_web_generic.png"
+                    width="210"
+                />
+            </a>
+            <a href="{{$ctrl.storeLinks[$ctrl.MOBILE_OS.windows]}}">
```
Comment (dectotam) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:32]:
```suggestion
            <a data-ng-href="{{$ctrl.storeLinks[$ctrl.MOBILE_OS.windows]}}">
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.constants.js (hunk 5 above)
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.constants.js:5]: I am not sure, I can find both 
image

**Review by dectotam — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.scss [hunk 8]
```diff
@@ -0,0 +1,19 @@
+@media (max-width: 991px) {
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.scss:1]: **nitpick:** Can you add comment that you're targeting mobile and tablets?
```suggestion
/* targeting mobile and tablet devices */
@media (max-width: 991px) {
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.scss
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.scss:1]: Also, remove the width from the HTML and add the `media-queries` for desktop screens.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.scss
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.scss:1]: I am afraid svg width property does not work in this case (does not resize)

**Review by anooparveti — APPROVED** [review]

**Pull request merged**